### PR TITLE
[ci skip] Remove incorrect documentation around git push -f

### LINKED ISSUE #1174: developer workflow is unclear about force pushing
The workflow suggests doing a force push on each commit.  During iteration of a PR, each separate change should be submitted instead, with a squash prior to merging.  See: https://github.com/istio/istio/pull/1061#discussion_r145501354

### PR DESCRIPTION
**Pull request by sdake**
Fixes #1174

Also add explination as to why force push shouldn't be used.

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```NONE
```

### CODE REVIEW

**Comment by ldemailly** [comment]
@yutongz any idea why we get errors on a text only pr?

**Comment by yutongz** [comment]
For circleci, no idea. For that prow job, because my PR https://github.com/istio/istio/pull/1525 is not merged, so there is not such file for Prow to find. Since there are some debate about if we need to enable auto-merge, should I remove that job (auto-merge_master) for now?

**Comment by ldemailly** [comment]
not sure what the debate is about, we need un attended merge for the project to scale, specially in mono repo

**Comment by yutongz** [comment]
Exactly. That's pretty clear to me, but apparently not everyone is on the same page.

**Comment by ldemailly** [comment]
@sdake can you resolve the conflict ? tia

**Comment by sdake** [comment]
@ldemailly should be gtg :)  thanks.

**Comment by yutongz** [comment]
/retest
Thanks!

File: DEV-GUIDE.md [hunk 1]
```diff
@@ -217,6 +217,12 @@ fixups (e.g. automated doc formatting), use one or more commits for the
 changes to tooling and a final commit to apply the fixup en masse. This makes
 reviews much easier.
 
+Please do not use force push after submitting a PR to resolve review
+feedback. Doing so results in an inability to see what has changed between
+revisions of the PR. Instead submit additional commits until the PR is
+suitable for merging. Once the PR is suitable for merging, the commits may
```
Comment (ldemailly) [inline DEV-GUIDE.md]: ^may^will

**Pull request merged**